multiversx_sdk_cli/contracts.py

```diff
@@ -61,161 +64,142 @@ class IContractQueryResponse(Protocol):
-        self.address = address
-        self.bytecode = bytecode
-        self.metadata = metadata or CodeMetadata()
-
-    def deploy(self, owner: Account, arguments: List[Any], gas_price: int, gas_limit: int, value: int, chain: str, version: int, guardian: str, options: int) -> Transaction:
-        self.owner = owner
-        address_computer = AddressComputer(number_of_shards=3)
-        self.address = address_computer.compute_contract_address(self.owner.address, self.owner.nonce)
-
-        arguments = arguments or []
-        gas_price = int(gas_price)
-        gas_limit = int(gas_limit)
-        value = value or 0
-
-        tx = Transaction(
-            chain_id=chain,
-            sender=owner.address.to_bech32(),
-            receiver=ADDRESS_ZERO_BECH32,
-            gas_limit=gas_limit,
-            gas_price=gas_price,
-            nonce=owner.nonce,
-            amount=value,
-            data=self.prepare_deploy_transaction_data(arguments).data,
-            version=version,
-            options=options
-        )
+class IConfig(Protocol):
+    chain_id: str
+    min_gas_limit: int
+    gas_limit_per_byte: int
 
-        if guardian:
-            tx.guardian = guardian
 
-        tx.signature = bytes.fromhex(owner.sign_transaction(tx))
-        return tx
+class IToken(Protocol):
+    identifier: str
+    nonce: int
 
-    def prepare_deploy_transaction_data(self, arguments: List[Any]) -> TransactionPayload:
-        tx_data = f"{self.bytecode}@{constants.VM_TYPE_WASM_VM}@{self.metadata.to_hex()}"
 
-        for arg in arguments:
-            tx_data += f"@{_prepare_argument(arg)}"
+class ITokenComputer(Protocol):
+    def is_fungible(self, token: IToken) -> bool:
+        ...
 
-        return TransactionPayload.from_str(tx_data)
+    def extract_identifier_from_extended_identifier(self, identifier: str) -> str:
+        ...
 
-    def execute(self, caller: Account, function: str, arguments: List[str], gas_price: int, gas_limit: int, value: int, chain: str, version: int, guardian: str, options: int) -> Transaction:
-        self.caller = caller
 
-        arguments = arguments or []
-        gas_price = int(gas_price)
-        gas_limit = int(gas_limit)
-        value = value or 0
-        receiver = self.address if self.address else EmptyAddress()
-
-        tx = Transaction(
-            chain_id=chain,
-            sender=caller.address.to_bech32(),
-            receiver=receiver.to_bech32(),
-            gas_limit=gas_limit,
-            gas_price=gas_price,
-            nonce=caller.nonce,
-            amount=value,
-            data=self.prepare_execute_transaction_data(function, arguments).data,
-            version=version,
-            options=options
+class SmartContract:
+    def __init__(self, config: IConfig, token_computer: ITokenComputer):
+        self._factory = SmartContractTransactionsFactory(config, token_computer)
+
+    def get_deploy_transaction(self, owner: Account, args: Any) -> Transaction:
```

Comment:
Yes, it's only used by `cli_contracts.py`. I thought it's a little bit "cleaner" to use `args` since `gasLimit`, `bytecode` and so on are required arguments, and we also check for the `nonce` and `chainID` args.